### PR TITLE
bugfix-603429

### LINKED ISSUE #65: Datetime2 is transferred as String
According to https://jtds.sourceforge.net/typemap.html, there is no mapping for type `datetime2`, hence it is read as String. Since the library doesn't perform any cast in this case, it is written as String as well. 
Would be good to have it mapped to timestamp.

### PR DESCRIPTION
**Pull request by VladOrlov**
resolves #65 

### Description

Convert datetime2 mssql column to timestamp type instead of string used by default

### Checklist

- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the README.md file

### CODE REVIEW

File: src/main/scala/mirroring/services/SparkContextTrait.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package mirroring.services
+
+import org.apache.spark.sql.SparkSession
+
+trait SparkContextTrait {
```
Comment (hanna-liashchuk) [inline src/main/scala/mirroring/services/SparkContextTrait.scala:20]: there is a SparkService with the same purpose, one of them should be removed

**Review by hanna-liashchuk — CHANGES_REQUESTED** [review]

File: src/main/scala/mirroring/services/SparkContextTrait.scala (hunk 1 above)
Comment (VladOrlov) [reply src/main/scala/mirroring/services/SparkContextTrait.scala:20]: SparkService removed

**Review by hanna-liashchuk — DISMISSED** [review]

File: src/main/scala/mirroring/Runner.scala [hunk 2]
```diff
@@ -34,62 +40,91 @@ object Runner extends LogSupport {
...
+
+    val loggerConfig = LoggerConfig(
+      schema = config.schema,
+      table = config.tab,
+      logLevel = config.logLvl,
+      applicationId = spark.sparkContext.applicationId,
+      applicationAttemptId = spark.sparkContext.applicationAttemptId.getOrElse("1")
+    )
+
+    FlowLogger.init(loggerConfig)
```
Comment (VShkaberda) [inline src/main/scala/mirroring/Runner.scala:80]: Moving `FlowLogger.init` from the line 48 breaks logging. E.g., for missing key now we got:

> [34m2023-06-12 12:01:50.292Z[0m [31merror[0m [[37mConfigBuilder[0m] [31mMissing value for key tab=VS_ct_test7[0m  [34m- (ConfigBuilder.scala:79)[0m
> Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 1 ...

Previous version log example:

> 23/06/12 12:01:30 INFO KubernetesClusterSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.8
> {"timestamp":"23/06/12 12:01:30","level":"ERROR","sparkApplicationId":"spark-e2cd1128e4fc4229b0e4103a4ba7f073","sparkApplicationAttempt":"1","mirrorTaskName":"mirroring_schema__tab","loggerName":"mirroring.builders.ConfigBuilder","message":"Missing value for key tab=VS_ct_test6"}
> 23/06/12 12:01:31 INFO SparkUI: Stopped Spark web UI at http://spark-a36e0888af7b442c-driver-svc.spark-jobs.svc:4040/
> 23/06/12 12:01:31 INFO KubernetesClusterSchedulerBackend: Shutting down all executors
> 23/06/12 12:01:31 INFO KubernetesClusterSchedulerBackend$KubernetesDriverEndpoint: Asking each executor to shut down
> 23/06/12 12:01:31 WARN ExecutorPodsWatchSnapshotSource: Kubernetes client has been closed.
> 23/06/12 12:01:32 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
> 23/06/12 12:01:32 INFO MemoryStore: MemoryStore cleared
> 23/06/12 12:01:32 INFO BlockManager: BlockManager stopped
> 23/06/12 12:01:32 INFO BlockManagerMaster: BlockManagerMaster stopped
> 23/06/12 12:01:32 INFO StatsdSink: StatsdSink stopped.
> 23/06/12 12:01:32 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
> 23/06/12 12:01:32 INFO SparkContext: Successfully stopped SparkContext
> Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 1 ...

File: src/main/scala/mirroring/services/databases/JdbcService.scala [hunk 3]
```diff
@@ -16,53 +16,59 @@
...
+      .format("jdbc")
+      .option("url", jdbcContext.url)
+      .option("customSchema", customSchema)
+  }
+
+  private def getCustomSchema: String = {
     // create custom schema to avoid transferring DATE as STRING
-    // viz https://jtds.sourceforge.net/typemap.html
-    val sql =
+    // viz https://jtds.sourceforge.net/typemap.htmlØ
```
Comment (VShkaberda) [inline src/main/scala/mirroring/services/databases/JdbcService.scala]: Symbol `Ø` at the end of line.

File: src/main/scala/mirroring/config/Config.scala [hunk 4]
```diff
@@ -73,7 +73,6 @@ case class Config(
     deletedFileRetentionDuration: String
 ) extends LogSupport {
 
-  FlowLogger.init(schema, tab, logLvl)
```
Comment (VShkaberda) [inline src/main/scala/mirroring/config/Config.scala:76]: Needed to reinitialize after initializing at the line 48 with dummy params.

**Review by VShkaberda — CHANGES_REQUESTED** [review]

**Review by VShkaberda — APPROVED** [review]

**Review by hanna-liashchuk — APPROVED** [review]

**Pull request merged**